src/Zynga/Framework/Lockable/Cache/V1/Driver/Caching.hh

```diff
@@ -58,37 +58,67 @@ class Caching extends FactoryDriverBase implements DriverInterface {
       throw $e;
     }
   }
-  
+
   public function isLocked(StorableObjectInterface $obj): bool {
 
     try {
       $lockKey = $this->getLockCacheKeyFromStorableObject($obj);
       $alreadyLocked = $this->_locks->get($lockKey);
-    
+
       // Check my thead already has a lock
       if ($alreadyLocked instanceof LockPayloadInterface) {
         // Check if the lock is still valid, if so we are done.
-        if ($alreadyLocked->isLockStillValid($this->getConfig()->getLockTTL())) {
+        if ($alreadyLocked->isLockStillValid(
+              $this->getConfig()->getLockTTL(),
+            )) {
           return true;
         }
       }
-      
+
       // Check if any other thread has a lock
       $lockCache = $this->getConfig()->getLockCache();
-      
+
       $lockPayload = new LockPayload();
 
       $lockPayload = $lockCache->get($lockPayload, $lockKey);
-      if($lockPayload === null) {
+      if ($lockPayload === null) {
         return false;
       }
-      
+
       return $lockPayload->isDefaultValue()[0];
-      
+
+    } catch (Exception $e) {
+      throw $e;
+    }
+
+  }
+
```

Comment:
thinking to keep both isLocked & isLockedByAnotherThread methods as utility just in case we need them for any future purpose. That said, no more required to check additional condition isLocked() before unlocking the resource. Now unlock gracefully checks that internally.

Reply:
Spoke to you in person, lets remove these for now as it adds confusion on where to use isLocked & isLockedByAnotherThread. 

So far in Pgdata we are not supporting lock conflicts within the same thread

Reply:
will remove this in next revision.